src/client/java/net/neoforged/neoforge/client/model/ao/EnhancedAoRenderStorage.java

```diff
@@ -48,9 +49,9 @@ public static void applyFlatQuadBrightness(BlockAndTintGetter level, BakedQuad q
             for (int vertex = 0; vertex < 4; ++vertex) {
                 // Handle each vertex separately to apply vertex normals.
 
-                int normal = quad.bakedNormals().normals(vertex);
+                int normal = quad.bakedNormals().normal(vertex);
                 // The ignored byte is padding and may be filled with user data
-                if ((normal & 0x00FFFFFF) == 0) {
+                if (BakedNormals.isUnspecified(normal)) {
                     // No normal! Try to use the quad normal.
                     if (quadNormal == -1) {
                         quadNormal = ClientHooks.computeQuadNormal(quad.position0(), quad.position1(), quad.position2(), quad.position3());
```

Comment:
Can we move `computeQuadNormal` to `BakedNormals` as well? And `ClientHooks.fillNormal` can likely be inlined.

Reply:
Yes very good idea. I missed a pack operation in there.